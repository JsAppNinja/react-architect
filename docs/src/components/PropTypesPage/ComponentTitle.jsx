import React from 'react';
import PropTypes from 'prop-types';
import Button from 'react-md/lib/Buttons/Button';
import TextField from 'react-md/lib/TextFields';
import TableCardHeader from 'react-md/lib/DataTables/TableCardHeader';

import { QuickLinkTitle } from 'components/QuickLink';

const ComponentTitle = ({
  mobile,
  id,
  source,
  component,
  propFilter,
  mobileFilterVisible,
  onFilter,
  onMobileFilterShow,
  onMobileFilterHide,
}) => {
  const filter = (
    <TextField
      id={`${id}-proptypes-filter`}
      key="filter"
      placeholder="Filter props"
      fullWidth={false}
      value={propFilter}
      onChange={onFilter}
    />
  );

  let actions;
  let children;
  if (mobile) {
    actions = <Button key="close" onClick={onMobileFilterHide} icon>arrow_back</Button>;
    children = <Button key="open" onClick={onMobileFilterShow} icon>filter_list</Button>;
  } else {
    children = [
      filter,
      <Button key="source" href={source} iconClassName="fa fa-github" icon />,
    ];
  }
  return (
    <TableCardHeader
      className="quick-link"
      title={<QuickLinkTitle id={`${id}-proptypes`} title={component} desktop={!mobile} />}
      titleId={`${id}-proptypes`}
      actions={actions}
      visible={mobileFilterVisible}
      contextualChildren={filter}
    >
      {children}
    </TableCardHeader>
  );
};

ComponentTitle.propTypes = {
  mobile: PropTypes.bool.isRequired,
  id: PropTypes.string.isRequired,
  source: PropTypes.string.isRequired,
  component: PropTypes.string.isRequired,
  propFilter: PropTypes.string.isRequired,
  mobileFilterVisible: PropTypes.bool.isRequired,
  onFilter: PropTypes.func.isRequired,
  onMobileFilterShow: PropTypes.func.isRequired,
  onMobileFilterHide: PropTypes.func.isRequired,
};

export default ComponentTitle;
