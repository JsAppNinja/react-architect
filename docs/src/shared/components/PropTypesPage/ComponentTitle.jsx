import React, { PureComponent, PropTypes } from 'react';
import CSSTransitionGroup from 'react-addons-css-transition-group';
import CardTitle from 'react-md/lib/Cards/CardTitle';
import Button from 'react-md/lib/Buttons/Button';
import TextField from 'react-md/lib/TextFields';
import Toolbar from 'react-md/lib/Toolbars';

export default class ComponentTitle extends PureComponent {
  static propTypes = {
    mobile: PropTypes.bool.isRequired,
    tablet: PropTypes.bool.isRequired,
    desktop: PropTypes.bool.isRequired,
    source: PropTypes.string.isRequired,
    component: PropTypes.string.isRequired,
    propFilter: PropTypes.string.isRequired,
    onPropFilter: PropTypes.func.isRequired,
    baseId: PropTypes.string.isRequired,
    style: PropTypes.object,
    className: PropTypes.string,
  };

  constructor(props) {
    super(props);

    this.state = {
      filtering: !props.mobile,
    };

    this._toggleFilter = this._toggleFilter.bind(this);
  }

  componentWillReceiveProps(nextProps) {
    if (this.props.mobile === nextProps.mobile) {
      return;
    }

    this.setState({ filtering: !nextProps.mobile });
  }

  _toggleFilter() {
    this.setState({ filtering: !this.state.filtering });
  }

  render() {
    const { filtering } = this.state;
    const {
      component,
      source,
      propFilter,
      onPropFilter,
      mobile,
      baseId,
    } = this.props;

    let filter;
    if (filtering) {
      filter = (
        <TextField
          id={`propTypesFilter-${component}`}
          placeholder="Filter properties"
          value={propFilter}
          onChange={onPropFilter}
          className="md-cell--right"
          fullWidth={false}
        />
      );
    }

    let toggle;
    if (mobile) {
      toggle = (
        <Button
          icon
          key="toggle"
          className="md-cell--right"
          onClick={this._toggleFilter}
          tooltipLabel="Show the filter toolbar"
          tooltipPosition="left"
        >
          filter_list
        </Button>
      );
    }

    if (filter && mobile) {
      const back = (
        <Button
          icon
          key="back"
          onClick={this._toggleFilter}
          tooltipLabel="Hide the filter toolbar"
          tooltipPosition="left"
        >
          arrow_back
        </Button>
      );

      filter = (
        <Toolbar key="toolbar" className="filter-toolbar" actions={back} themed>
          {filter}
        </Toolbar>
      );
    }

    return (
      <CSSTransitionGroup
        id={`${baseId}-proptypes`}
        component={CardTitle}
        transitionName="md-drop-down"
        transitionEnterTimeout={150}
        transitionLeaveTimeout={150}
        title={component}
        style={{ position: 'relative', overflow: 'hidden' }}
      >
        {toggle}
        {filter}
        <Button
          icon
          forceIconSize
          href={source}
          iconClassName="fa fa-github"
          tooltipLabel={`Github source for ${component}`}
          tooltipDelay={300}
          tooltipPosition="left"
        />
      </CSSTransitionGroup>
    );
  }
}
