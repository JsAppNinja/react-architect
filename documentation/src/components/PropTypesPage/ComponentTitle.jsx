import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import Button from 'react-md/lib/Buttons/Button';
import TextField from 'react-md/lib/TextFields';
import TableCardHeader from 'react-md/lib/DataTables/TableCardHeader';

export default class ComponentTitle extends PureComponent {
  static propTypes = {
    id: PropTypes.string.isRequired,
    mobile: PropTypes.bool.isRequired,
    source: PropTypes.string.isRequired,
    component: PropTypes.string.isRequired,
    propFilter: PropTypes.string.isRequired,
    onFilter: PropTypes.func.isRequired,
  };

  constructor(props) {
    super(props);

    this.state = { visible: false };
  }

  open = () => {
    this.setState({ visible: true });
  };

  close = () => {
    this.setState({ visible: false });
  };

  render() {
    const { visible } = this.state;
    const { id, source, component, mobile, propFilter, onFilter } = this.props;

    const filter = (
      <TextField
        id={`${id}-proptypes-filter`}
        key="filter"
        placeholder="Filter props"
        fullWidth={false}
        value={propFilter}
        onChange={onFilter}
      />
    );

    let actions;
    let children;
    if (mobile) {
      actions = <Button key="close" onClick={this.close} icon>arrow_back</Button>;
      children = <Button key="open" onClick={this.open} icon>filter_list</Button>;
    } else {
      children = [
        filter,
        <Button key="source" href={source} iconClassName="fa fa-github" icon />,
      ];
    }
    return (
      <TableCardHeader
        title={component}
        titleId={`${id}-proptypes`}
        actions={actions}
        visible={visible}
      >
        {children}
      </TableCardHeader>
    );
  }
}
