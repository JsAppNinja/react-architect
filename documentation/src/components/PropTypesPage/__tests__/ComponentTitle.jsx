/* eslint-env jest */
import React from 'react';
import renderer from 'react-test-renderer';

import ComponentTitle from '../ComponentTitle';

jest.mock('react-md/lib/TextFields/TextField');

const PROPS = {
  id: 'autocompletes',
  component: 'Autocomplete',
  mobile: false,
  source: 'https://github.com/malursen/react-md/master/tree/src/js/Autocompletes/Autocomplete.js',
  propFilter: '',
  onFilter: () => {},
};
describe('ComponentTitle', () => {
  it('should render correctly', () => {
    const tree1 = renderer.create(<ComponentTitle {...PROPS} />).toJSON();
    const tree2 = renderer.create(<ComponentTitle {...PROPS} mobile />).toJSON();
    expect(tree1).toMatchSnapshot();
    expect(tree2).toMatchSnapshot();
  });
});
