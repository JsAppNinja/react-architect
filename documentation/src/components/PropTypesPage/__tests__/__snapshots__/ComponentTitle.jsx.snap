// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ComponentTitle should render correctly 1`] = `
<header
  className="md-table-card-header"
  style={undefined}
>
  <div
    className="md-card-title"
  >
    <h2
      className="md-card-title--title md-text md-card-title--large"
      id="autocompletes-proptypes"
    >
      Autocomplete
    </h2>
    <a
      className="md-inline-block md-btn md-btn--icon md-pointer--hover"
      disabled={undefined}
      href="https://github.com/malursen/react-md/master/tree/src/js/Autocompletes/Autocomplete.js"
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
    >
      <i
        className="md-icon fa fa-github"
        style={undefined}
      />
    </a>
  </div>
</header>
`;

exports[`ComponentTitle should render correctly 2`] = `
<header
  className="md-table-card-header"
  style={undefined}
>
  <div
    className="md-card-title"
  >
    <h2
      className="md-card-title--title md-text md-card-title--large"
      id="autocompletes-proptypes"
    >
      Autocomplete
    </h2>
    <button
      className="md-inline-block md-btn md-btn--icon md-pointer--hover md-cell--right"
      disabled={undefined}
      href={undefined}
      onClick={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
      type="button"
    >
      <i
        className="md-icon material-icons"
        style={undefined}
      >
        filter_list
      </i>
    </button>
  </div>
</header>
`;
